src/components/content/WithPlaceholder/WithPlaceholder.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactNode } from 'react'
+
+export type WithPlaceholderProps = {
+  /**
+   * Defines the children to be rendered.
+   */
+  children: ReactNode
+  /**
+   * Defines the placeholder to be rendered if the children is not valid.
+   */
+  placeholder?: ReactNode
+}
```

Comment:
```suggestion
export type WithPlaceholderProps = PropsWithChildren<{
  /**
   * Defines the placeholder to be rendered if the children is not valid.
   */
  placeholder?: ReactNode
}>
```

React provides already a type definition for children.